pkg/shipper/shipper.go

```diff
@@ -304,7 +313,13 @@ func (s *Shipper) Sync(ctx context.Context) (uploaded int, err error) {
 		// Check against bucket if the meta file for this block exists.
 		ok, err := s.bucket.Exists(ctx, path.Join(m.ULID.String(), block.MetaFilename))
 		if err != nil {
-			return 0, errors.Wrap(err, "check exists")
+			if !s.skipCorruptedBlocks {
+				s.metrics.dirSyncFailures.Inc()
+				return uploaded, errors.Wrapf(err, "check exists - ulid: %s", m.ULID.String())
+			}
+			failedBlocks = append(failedBlocks, m.ULID.String())
+			level.Warn(s.logger).Log("msg", "check exists failure", "err", err, "ULID", m.ULID.String())
```

Comment:
Should we leave this as is? If it failed to check if the block meta exists on S3 or not, it has nothing to do with corrupted blocks on disk.

I thought `skipCorruptedBlocks` is for skipping block on disk only

Reply:
I misunderstood the code. This flag it not really what we need here, but it would also make sense to continue execution even if it found a block in the bucket already. Not sure if we want to just change this behave without a need of a flag. If other blocks are already in the bucket it would skip as well